_posts/2018-02-16-Stairway-To-Health-2.md

```diff
@@ -0,0 +1,344 @@
+- Upgraded from Angular 4 to Angular 5
+- Material Design
+- NEST.js in stead of Express.js (still Express underneath, but cleaner code!)
+- Backend e2e tests with Mockgoose
+- Deployed on OpenShift
+- New type of sensors
+- Cheers feature, users can motivate and support each other
+
+
+<h2>Stairway to Health @ Ordina</h2>
+As you might have read in our [previous post](https://ordina-jworks.github.io/iot/2017/10/12/Stairway-To-Health.html) about Stairway to Health, the purpose of the application is to improve worker health in a fun and engaging way. 
+With the app we try to encourage employees to take the stairs instead of the elevator.
+We've put up some sensors that can detect how much the stairs are used on a per floor basis and how many people take the elevator.
+In the app they can see the results and thus they can do an extra effort if they are falling behind.
+New in the Ordina version is that employees can now also cheer and motivate each other since we've added a chat feature to the application.
+
+
+<h2>Internet of Things</h2>
+The [Stairway to Health](http://sth2-fe-prod-ordina-stairway-to-health.origin.ordina-jworks.io/dashboard) project is a simple yet great example to show what the Internet of Things can do:
+- LoRa sensors detect door openings, these are installed on the doors of the staircases
+- These sensors communicate via the LoRa network to report their status
+- In our case sensor data is sent to the Proximus MyThings platform which processes the data
+- The data gets sent to the Stairway to Health application
+- The Stairway to Health application interprets and visualises the data
+
+In summary: We install sensors on the doors (things) to measure usage and we analyse the data to persuade people to move more.
+The result is a good example of how IoT can influence our daily lives.
+
+For more on this topic, check the application's [About page](http://sth2-fe-prod-ordina-stairway-to-health.origin.ordina-jworks.io/about)</a>
+
+
+<h2>Dive into the technical details</h2>
+The reason of us writing this blog post is mainly because we want to explain some of the technical changes and improvements we've made
+since we've updated (pretty much rewritten) the application. 
+So let's get started.
+
+
+<h3>The API</h3>
+ExpressJs to NestJs: The main difference here is that we've rewritten the application to use the new framework inf favour of the old implementation with ExpressJs.
+Migrating from Express to Nest is not that difficult, since nest is a wrapper on top of the Express framework.
+It provides you with some nice TypeScript decorators which makes your code a lot cleaner, more compact and easier to read.
+
+
+<h4>ExpressJs example</h4>
+```typescript
+export class EntityApi extends CoreApi {
+    private entityController: EntityController = new EntityController();
+    constructor() {
+        super();
+    }
+
+    // the create function would that have to be executed by the main server while bootstrapping the application
```

Comment:
If possible try to reduce the length of this comment a tiny bit because there's currently a scroll bar due to its length.